runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java

```diff
@@ -84,19 +99,74 @@
 	@Inject
 	private ChatboxPanelManager chatboxPanelManager;
 
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private FriendNotesConfig config;
+
 	@Getter
 	private HoveredFriend hoveredFriend = null;
 
+	private int iconIdx = -1;
+	private String currentlyLayouting = null;
+
+	@Provides
+	private FriendNotesConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(FriendNotesConfig.class);
+	}
+
 	@Override
 	protected void startUp() throws Exception
 	{
 		overlayManager.add(overlay);
+		loadIcon();
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			rebuildFriendsList();
+			rebuildIgnoreList();
+		}
 	}
 
 	@Override
 	protected void shutDown() throws Exception
 	{
 		overlayManager.remove(overlay);
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			rebuildFriendsList();
+			rebuildIgnoreList();
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() == GameState.LOGGED_IN)
+		{
+			loadIcon();
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals(CONFIG_GROUP))
+		{
+			return;
+		}
+
+		if (!event.getKey().equals("showIcons"))
+		{
+			return;
+		}
+
+		if (client.getGameState() == GameState.LOGGED_IN)
```

Comment:
These conditions can all be combined.
```suggestion
		if (event.getGroup().equals(CONFIG_GROUP) && event.getKey().equals("showIcons") && client.getGameState() == GameState.LOGGED_IN)
```

Reply:
easier to extend when its like this imo

Reply:
replaced with a switch statement for future expansion if wanted